lib/web_ui/dev/driver_manager.dart

```diff
@@ -0,0 +1,145 @@
+import 'package:path/path.dart' as pathlib;
+import 'package:web_driver_installer/chrome_driver_installer.dart';
+import 'package:web_driver_installer/safari_driver_runner.dart';
+
+import 'chrome_installer.dart';
+import 'common.dart';
+import 'environment.dart';
+import 'utils.dart';
+
+/// [DriverManager] implementation for Chrome.
+///
+/// This manager can be used for both MacOS and Linux.
+class ChromeDriverManager extends DriverManager {
+  ChromeDriverManager(String browser) : super(browser);
+
+  Future<void> _installDriver() async {
+    if (_browserDriverDir.existsSync()) {
+      _browserDriverDir.deleteSync(recursive: true);
+    }
+
+    _browserDriverDir.createSync(recursive: true);
+    temporaryDirectories.add(_drivers);
+
+    io.Directory temp = io.Directory.current;
+    io.Directory.current = _browserDriverDir;
+
+    // TODO(nurhan): https://github.com/flutter/flutter/issues/53179
+    final String chromeDriverVersion = await queryChromeDriverVersion();
+    ChromeDriverInstaller chromeDriverInstaller =
+        ChromeDriverInstaller.withVersion(chromeDriverVersion);
+    await chromeDriverInstaller.install(alwaysInstall: true);
+    io.Directory.current = temp;
+  }
+
+  /// Driver should already exist on LUCI as a CIPD package.
+  ///
+  /// Throw an error if directory does not exists.
+  Future<void> _verifyDriverForLUCI() {
+    if (!_browserDriverDir.existsSync()) {
+      throw StateError('Failed to locate Chrome driver on LUCI on path:'
+          '${_browserDriverDir.path}');
+    }
+    return Future<void>.value();
+  }
+
+  Future<void> _startDriver(String driverPath) async {
+    await startProcess('./chromedriver/chromedriver', ['--port=4444'],
+        workingDirectory: driverPath);
+    print('INFO: Driver started');
+  }
+}
+
+/// [DriverManager] implementation for Safari.
+///
+/// This manager is will only be created/used for MacOS.
+class SafariDriverManager extends DriverManager {
+  SafariDriverManager(String browser) : super(browser);
+
+  Future<void> _installDriver() {
+    // No-op.
+    // macOS comes with Safari Driver installed.
+    return new Future<void>.value();
+  }
+
+  Future<void> _verifyDriverForLUCI() {
+    // No-op.
+    // macOS comes with Safari Driver installed.
+    return Future<void>.value();
+  }
+
+  Future<void> _startDriver(String driverPath) async {
+    final SafariDriverRunner safariDriverRunner = SafariDriverRunner();
+
+    final io.Process process =
+        await safariDriverRunner.runDriver(version: 'system');
+
+    processesToCleanUp.add(process);
+  }
+}
+
+/// Abstract class for preparing the browser driver before running the integration
+/// tests.
+abstract class DriverManager {
+  /// Installation directory for browser's driver.
+  ///
+  /// Always re-install since driver can change frequently.
+  /// It usually changes with each the browser version changes.
+  /// A better solution would be installing the browser and the driver at the
+  /// same time.
```

Comment:
Should the second paragraph be on another method, e.g. `_installDriver`, or perhaps the class itself? I doubt I'd be looking for information about how often we install the driver on a field called `_browserDriverDir`.

Reply:
done